src/core/public/application/application_service.tsx

```diff
@@ -304,14 +311,18 @@ export class ApplicationService {
         return absolute ? relativeToAbsolute(relUrl) : relUrl;
       },
       navigateToApp,
-      navigateToUrl: async (url) => {
+      navigateToUrl: async (url: string, { skipAppLeave }: NavigateToUrlOptions = {}) => {
         const appInfo = parseAppUrl(url, http.basePath, this.apps);
         if (appInfo) {
-          return navigateToApp(appInfo.app, { path: appInfo.path });
+          return navigateToApp(appInfo.app, { path: appInfo.path, skipAppLeave });
         } else {
           return this.redirectTo!(url);
         }
       },
+      navigateToUrlSkipUnload: (url: string) => {
+        window.removeEventListener('beforeunload', this.onBeforeUnload);
+        return this.redirectTo!(url);
+      },
```

Comment:
I'm sorry to suggest that after being the one that explicitly asked for a dedicated method, but looking again at the `navigateToUrl` implementation, and given the fact that this PR introduces the `skipAppLeave` option for it, I wonder if we shouldn't just add an additional `forceRedirect` option `navigateToUrl` (and maybe `navigateToApp` too)

This `forceRedirect` option would force `navigateToUrl` to use `this.redirectTo` instead of `navigateToApp`, even if the url matched an existing application.

The security logout hook could then use both those options to answer their usecase:

```ts
core.application.navigateToUrl(logoutUrl, { skipAppLeave: true, forceRedirect: true})
```

WDYT?

Reply:
@jportner would using the `skipAppLeave` flag in `navigateToUrl` work for you? I ask because we originally thought to add a new nav method specifically to handle the session timeout edge case.

Reply:
@pgayvallet Joe and I tested your recommendation out and it seems to work. It does raise one question though: would we still need _both_ `skipAppLeave` and `forceRedirect` flags in `navigateToUrl`?
On second thought though, we do need `skipAppLeave` in `navigateToApp` to handle potential deeplinks that we might be redirecting to, right?

Reply:
I like Pierre's idea here too, and I think it answers my initial concern that we should "prefer one way of doing things"

The existing implementation with the dedicated method + new `navigateToUrl` params feels like two ways of accomplishing the same thing, so I'm +1 to either dropping the dedicated method, or making the dedicated method broader in scope so that it serves a different purpose (as mentioned in the other thread).

Dropping the dedicated method in favor of the new options on the existing API is probably the simplest approach in the end.

Reply:
I've made the changes after confirming with security that the flag option works too. We're going with:
```
core.application.navigateToUrl(logoutUrl, { skipAppLeave: true, forceRedirect: true})
```